src/components/TaskProjectSelector.vue

```diff
@@ -0,0 +1,283 @@
+<template>
+  <div class="projects">
+    <label for="projects">Projects</label>
+    <div class="project-list">
+      <div
+        class="project"
+        v-for="membership in taskMemberships"
+        :key="membership.project.gid"
+      >
+        <span class="column even">{{ membership.project.name }}</span>
+        <span class="column" v-if="membership.section">{{
+          getSwimlane(membership.section.name)
+        }}</span>
+        <span class="column even" v-if="membership.section">{{
+          getSection(membership.section.name)
+        }}</span>
+        <n-icon
+          class="trash column"
+          @click="removeMembership(taskId, membership.project.gid)"
+        >
+          <trash-can />
+        </n-icon>
+      </div>
+    </div>
+    <div class="task-project-selector">
+      <n-button
+        @click="
```

Comment:
Generally you want the event handlers to just call a function in the component (rather than inline the code). Makes it easier to reason and debug